CADETPythonSimulator/solver.py

```diff
@@ -0,0 +1,292 @@
+            The time points at which the solution is sought.
+        #TODO: Consider creating a section class instead of using Addict
+
+        """
+        self._update_unit_operation_parameters(
+            section.start,
+            section.end,
+            section.section_states,
+        )
+        self._system.update_system_connectivity(section.connections)
+
+        section_solution_times = self.get_section_solution_times(section)
+
+        y_initial = self._system.y
+        y_initial_dot = self._system.y_dot
+        output = self.solver.solve(
+            section_solution_times, y_initial, y_initial_dot
+        )
+
+        y_history = output.values.y
+        y_dot_history = output.values.ydot
+        times = output.values.t
+
+        self._system.y = y_history[-1]
+        self._system.y_dot = y_dot_history[-1]
+
+        self.write_solution(times, y_history, y_dot_history)
+
+    def get_section_solution_times(self, section: Dict) -> np.ndarray:
+        """
+        Calculate Section Times.
+
+        Every Section describes the solution times with a start and end time.w
+        The solution times are simply calculated by using np.arange with time_resolution
+        as step size. If time_resolution is not given, a resolution of 1s is used.
+
+        Parameters
+        ----------
+        section : Dict
+            Section Dict that describes the Parameters
+
+        """
+        # TODO: How to get section_solution_times from section.start, section.end, if 
+        # user_solution times are provided?
+        time_resolution = section.get("time_resolution", 1.0)
+        start = section.start
+        end = section.end
+        return np.arange(start, end, time_resolution)
+
+
+
+    def _update_unit_operation_parameters(
+            self,
+            start: float,
+            end: float,
+            unit_operation_parameters: dict[
+                UnitOperationBase | str,
+                dict[str, npt.ArrayLike]
+                ]
+            ) -> np.ndarray:
+        """
+        Update time dependent unit operation parameters.
+
+        Parameters
+        ----------
+        start: float
+            Start time of the section.
+        end: float
+            End time of the section.
+        unit_operation_parameters : dict[UnitOperation | str, dict]
+            Unit operation parameters for the next section.
+
+        """
+        for unit, parameters in unit_operation_parameters.items():
+            if isinstance(unit, str):
+                try:
+                    unit = self._system.unit_operations[unit]
+                except KeyError:
+                    raise CADETPythonSimError(f"Unit {unit} is not Part of the System.")
+            else:
+                if unit not in self._system.unit_operations.values():
+                    raise CADETPythonSimError(f"Unit {unit} is not Part of the System.")
+
+            unit.update_parameters(start, end, parameters)
+
+    # @property
```

Comment:
do we still need this?

Reply:
No, port mapping is a leftover from the splitting and the old structure. I will remove it.